### ISSUE TITLE
[REGRESSION]: Class WebSwapCGLLayer is implemented in both <paths>. One of the two will be used. Which one is undefined.

### ISSUE THREAD

**Issue by worldomonation** [microsoft/playwright#17208]
**Context:**
- GOOD Playwright Version: [what Playwright version worked nicely?] 1.23.x
- BAD Playwright Version: [what Playwright version doesn't work any more?] 1.24.x
- Operating System: [e.g. Windows, Linux or Mac] macOS 12.5.1 
- Extra: [any specific details about your environment] nodeJS 18.4.0 (system installed), Jest 27.5.1

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```typescript
import { Browser, Page, chromium } from 'playwright';

describe( 'Minimal', function () {
	let page1: Page;
	let browser: Browser;

	it( 'Setup browser', async function () {
		browser = await chromium.launch( { headless: false } );
		page1 = await browser.newPage();
	} );

	it( 'Using page1, open a site', async function () {
		await page1.goto( 'https://wordpress.com' );
	} );

	it( 'Open page2 and navigate to a site', async function () {
		const page2 = await browser.newPage();

		await page2.goto( 'https://wordpress.com' );
		await page2.close();
	} );

	it( 'Crashing step', async function () {
		const locator = page1.locator( 'some-selector' );
		await locator.waitFor();
	} );
} );

```

Note: the site does _not_ have to be `wordpress.com` - it can be any other site eg. `google.ca`.

**Describe the bug**

When there is a persistent (in the context of the duration of the test script) if a new browser page is launched then subsequently closed, **and then** the test script attempts to wait for a locator on the original page, then the browser crashes (log below).

**Log**

```
    locator.waitFor: Browser closed.
    ==================== Browser output: ====================
    <launching> /Users/edwintakahashi/Library/Caches/ms-playwright/chromium-1019/chrome-mac/Chromium.app/Contents/MacOS/Chromium --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --enable-use-zoom-for-dsf=false --no-sandbox --user-data-dir=/var/folders/2t/d2z_1zks2cq_wn2zn8whfxy80000gn/T/playwright_chromiumdev_profile-bpWXt7 --remote-debugging-pipe --no-startup-window
    <launched> pid=14139
    [pid=14139][err] objc[14148]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x245321b50) and /Users/edwintakahashi/Library/Caches/ms-playwright/chromium-1019/chrome-mac/Chromium.app/Contents/Frameworks/Chromium Framework.framework/Versions/105.0.5195.19/Libraries/libGLESv2.dylib (0x1052ae3b0). One of the two will be used. Which one is undefined.
    [pid=14139][err] [14139:259:0908/171514.456335:FATAL:profile_destroyer.cc(266)] Check failed: !observations_.IsObservingAnySource(). Some render process hosts were not destroyed early enough!
    [pid=14139][err] 0   Chromium Framework                  0x0000000115045598 base::debug::CollectStackTrace(void**, unsigned long) + 28
    [pid=14139][err] 1   Chromium Framework                  0x0000000114fadf4c base::debug::StackTrace::StackTrace() + 24
    [pid=14139][err] 2   Chromium Framework                  0x0000000114fbf9f8 logging::LogMessage::~LogMessage() + 156
    [pid=14139][err] 3   Chromium Framework                  0x0000000114fc05c4 logging::LogMessage::~LogMessage() + 12
    [pid=14139][err] 4   Chromium Framework                  0x0000000114fa9498 logging::CheckError::~CheckError() + 36
    [pid=14139][err] 5   Chromium Framework                  0x0000000114d16a54 ProfileDestroyer::~ProfileDestroyer() + 228
    [pid=14139][err] 6   Chromium Framework                  0x0000000114d16eac ProfileDestroyer::~ProfileDestroyer() + 12
    [pid=14139][err] 7   Chromium Framework                  0x0000000111cb3c80 base::internal::Invoker<base::internal::BindState<void (net::(anonymous namespace)::DnsHTTPAttempt::*)(), base::WeakPtr<net::(anonymous namespace)::DnsHTTPAttempt>>, void ()>::RunOnce(base::internal::BindStateBase*) + 132
    [pid=14139][err] 8   Chromium Framework                  0x0000000115036de0 base::OneShotTimer::FireNow() + 52
    [pid=14139][err] 9   Chromium Framework                  0x000000011501a854 base::DefaultDelayedTaskHandleDelegate::RunTask(base::OnceCallback<void ()>) + 48
    [pid=14139][err] 10  Chromium Framework                  0x00000001122d36e4 void base::internal::FunctorTraits<void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), void>::Invoke<void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), base::WeakPtr<media::PipelineImpl::RendererWrapper>, base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>>(void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), base::WeakPtr<media::PipelineImpl::RendererWrapper>&&, base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>&&) + 132
    [pid=14139][err] 11  Chromium Framework                  0x00000001150025ec base::TaskAnnotator::RunTaskImpl(base::PendingTask&) + 272
    [pid=14139][err] 12  Chromium Framework                  0x00000001150168d0 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*) + 644
    [pid=14139][err] 13  Chromium Framework                  0x0000000115016510 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() + 100
    [pid=14139][err] 14  Chromium Framework                  0x0000000115055968 base::MessagePumpCFRunLoopBase::RunWork() + 104
    [pid=14139][err] 15  Chromium Framework                  0x000000011505058c base::mac::CallWithEHFrame(void () block_pointer) + 16
    [pid=14139][err] 16  Chromium Framework                  0x00000001150550f8 base::MessagePumpCFRunLoopBase::RunDelayedWorkTimer(__CFRunLoopTimer*, void*) + 68
    [pid=14139][err] 17  CoreFoundation                      0x00000001b60f7964 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32
    [pid=14139][err] 18  CoreFoundation                      0x00000001b60f754c __CFRunLoopDoTimer + 1064
    [pid=14139][err] 19  CoreFoundation                      0x00000001b60f6fd4 __CFRunLoopDoTimers + 392
    [pid=14139][err] 20  CoreFoundation                      0x00000001b60dba64 __CFRunLoopRun + 1936
    [pid=14139][err] 21  CoreFoundation                      0x00000001b60dab34 CFRunLoopRunSpecific + 600
    [pid=14139][err] 22  HIToolbox                           0x00000001bed1a338 RunCurrentEventLoopInMode + 292
    [pid=14139][err] 23  HIToolbox                           0x00000001bed1a0b4 ReceiveNextEventCommon + 564
    [pid=14139][err] 24  HIToolbox                           0x00000001bed19e68 _BlockUntilNextEventMatchingListInModeWithFilter + 72
    [pid=14139][err] 25  AppKit                              0x00000001b8c4251c _DPSNextEvent + 860
    [pid=14139][err] 26  AppKit                              0x00000001b8c40e14 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1328
    [pid=14139][err] 27  Chromium Framework                  0x0000000114bbab4c __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke + 64
    [pid=14139][err] 28  Chromium Framework                  0x000000011505058c base::mac::CallWithEHFrame(void () block_pointer) + 16
    [pid=14139][err] 29  Chromium Framework                  0x0000000114bbaab0 -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 128
    [pid=14139][err] 30  AppKit                              0x00000001b8c32fe0 -[NSApplication run] + 596
    [pid=14139][err] 31  Chromium Framework                  0x00000001150560a8 base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 312
    [pid=14139][err] 32  Chromium Framework                  0x0000000115054c58 base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 216
    [pid=14139][err] 33  Chromium Framework                  0x00000001150170a0 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) + 320
    [pid=14139][err] 34  Chromium Framework                  0x0000000114fe7c10 base::RunLoop::Run(base::Location const&) + 464
    [pid=14139][err] 35  Chromium Framework                  0x0000000112ca3c20 content::BrowserMainLoop::RunMainMessageLoop() + 180
    [pid=14139][err] 36  Chromium Framework                  0x0000000112ca5320 content::BrowserMainRunnerImpl::Run() + 24
    [pid=14139][err] 37  Chromium Framework                  0x0000000112ca19ac content::BrowserMain(content::MainFunctionParams) + 200
    [pid=14139][err] 38  Chromium Framework                  0x0000000114b34db4 content::RunBrowserProcessMain(content::MainFunctionParams, content::ContentMainDelegate*) + 156
    [pid=14139][err] 39  Chromium Framework                  0x0000000114b35fcc content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams, bool) + 456
    [pid=14139][err] 40  Chromium Framework                  0x0000000114b35d90 content::ContentMainRunnerImpl::Run() + 820
    [pid=14139][err] 41  Chromium Framework                  0x0000000114b34854 content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) + 2308
    [pid=14139][err] 42  Chromium Framework                  0x0000000114b3498c content::ContentMain(content::ContentMainParams) + 84
    [pid=14139][err] 43  Chromium Framework                  0x0000000111b94cdc ChromeMain + 256
    [pid=14139][err] 44  Chromium                            0x0000000102018938 main + 288
    [pid=14139][err] 45  dyld                                0x00000001022f908c start + 520
    [pid=14139][err] Task trace:
    [pid=14139][err] 0   Chromium Framework                  0x0000000114d165c0 ProfileDestroyer::ProfileDestroyer(Profile*, std::Cr::set<content::RenderProcessHost*, std::Cr::less<content::RenderProcessHost*>, std::Cr::allocator<content::RenderProcessHost*>> const&) + 380
    [pid=14139][err] 1   Chromium Framework                  0x0000000117574820 DevToolsBrowserContextManager::OnBrowserRemoved(Browser*) + 296
    [pid=14139][err] 2   Chromium Framework                  0x0000000116876fac remote_cocoa::NativeWidgetNSWindowBridge::CloseWindow() + 472
    [pid=14139][err] 3   Chromium Framework                  0x00000001159b99bc IPC::(anonymous namespace)::ChannelAssociatedGroupController::Accept(mojo::Message*) + 1092
    [pid=14139][err] 4   Chromium Framework                  0x00000001151b8060 mojo::SimpleWatcher::Context::Notify(unsigned int, MojoHandleSignalsState, unsigned int) + 288
    [pid=14139][err] Crash keys:
    [pid=14139][err]   "total-discardable-memory-allocated" = "4194304"
    [pid=14139][err]   "ax_mode" = "kNativeAPIs | kWebContents | kInlineTextBoxes | kScreenReader | "
    [pid=14139][err]   "gpu-glver" = "OpenGL ES 2.0.0 (ANGLE 2.1.19112 git hash: d0fe12db855d)"
    [pid=14139][err]   "gpu-generation-intel" = "0"
    [pid=14139][err]   "gpu-vsver" = "1.00"
    [pid=14139][err]   "gpu-psver" = "1.00"
    [pid=14139][err]   "gpu-driver" = "76.3"
    [pid=14139][err]   "gpu-devid" = "0x0000"
    [pid=14139][err]   "gpu-venid" = "0x106b"
    [pid=14139][err]   "ui_scheduler_async_stack" = "0x114D165C0 0x117574820"
    [pid=14139][err]   "num-extensions" = "0"
    [pid=14139][err]   "amfi-status" = "rv=0 status=0x0 allow_everything=0"
    [pid=14139][err]   "io_scheduler_async_stack" = "0x112C3B3CC 0x114D165C0"
    [pid=14139][err]   "variations" = "b357b792-3f4a17df,9481ce98-3d47f4f4,4b935545-3d47f4f4,70678518-dee66fa8,9a38bae3-3d47f4f4,c9af35af-ca79f5c6,122c746b-3d47f4f4,b0455224-3d47f4f4,d4e31d2-3d47f4f4,d69d967d-3695c92e,"
    [pid=14139][err]   "num-experiments" = "10"
    [pid=14139][err]   "switch-33" = "--no-startup-window"
    [pid=14139][err]   "switch-32" = "--remote-debugging-pipe"
    [pid=14139][err]   "switch-31" = "--user-data-dir=/var/folders/2t/d2z_1zks2cq_wn2zn8whfxy80000gn/T"
    [pid=14139][err]   "switch-30" = "--no-sandbox"
    [pid=14139][err]   "switch-29" = "--enable-use-zoom-for-dsf=false"
    [pid=14139][err]   "switch-28" = "--export-tagged-pdf"
    [pid=14139][err]   "switch-27" = "--no-service-autorun"
    [pid=14139][err]   "switch-26" = "--use-mock-keychain"
    [pid=14139][err]   "switch-25" = "--password-store=basic"
    [pid=14139][err]   "switch-24" = "--enable-automation"
    [pid=14139][err]   "switch-23" = "--no-first-run"
    [pid=14139][err]   "switch-22" = "--metrics-recording-only"
    [pid=14139][err]   "switch-21" = "--force-color-profile=srgb"
    [pid=14139][err]   "switch-20" = "--disable-sync"
    [pid=14139][err]   "switch-19" = "--disable-renderer-backgrounding"
    [pid=14139][err]   "switch-18" = "--disable-prompt-on-repost"
    [pid=14139][err]   "switch-17" = "--disable-popup-blocking"
    [pid=14139][err]   "switch-16" = "--disable-ipc-flooding-protection"
    [pid=14139][err]   "switch-15" = "--disable-hang-monitor"
    [pid=14139][err]   "switch-14" = "--allow-pre-commit-input"
    [pid=14139][err]   "switch-13" = "--disable-features=ImprovedCookieControls,LazyFrameLoading,Globa"
    [pid=14139][err]   "switch-12" = "--disable-extensions"
    [pid=14139][err]   "switch-11" = "--disable-dev-shm-usage"
    [pid=14139][err]   "switch-10" = "--disable-default-apps"
    [pid=14139][err]   "switch-9" = "--disable-component-extensions-with-background-pages"
    [pid=14139][err]   "switch-8" = "--disable-client-side-phishing-detection"
    [pid=14139][err]   "switch-7" = "--disable-breakpad"
    [pid=14139][err]   "switch-6" = "--disable-back-forward-cache"
    [pid=14139][err]   "switch-5" = "--disable-backgrounding-occluded-windows"
    [pid=14139][err]   "switch-4" = "--disable-background-timer-throttling"
    [pid=14139][err]   "switch-3" = "--enable-features=NetworkService,NetworkServiceInProcess"
    [pid=14139][err]   "switch-2" = "--disable-background-networking"
    [pid=14139][err]   "switch-1" = "--disable-field-trial-config"
    [pid=14139][err]   "num-switches" = "35"
    [pid=14139][err]   "osarch" = "arm64"
    [pid=14139][err]   "pid" = "14139"
    [pid=14139][err]   "ptype" = "browser"
    [pid=14139][err]
    [pid=14139][err] [0908/171514.678895:WARNING:crash_report_exception_handler.cc(235)] UniversalExceptionRaise: (os/kern) failure (5)
    [pid=14139] <process did exit: exitCode=null, signal=SIGTRAP>
    [pid=14139] starting temporary directories cleanup
    =========================== logs ===========================
    waiting for selector "iframe.is-loaded" to be visible
    ============================================================

      23 | 	it( 'Crashing step', async function () {
      24 | 		const locator = page1.locator( 'iframe.is-loaded' );
    > 25 | 		await locator.waitFor();
         | 		              ^
      26 | 	} );
      27 | } );
      28 |

      at Object.<anonymous> (specs/editor/editor__super_minimal.ts:25:17)
      at TestScheduler.scheduleTests (../../node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (../../node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (../../node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (../../node_modules/@jest/core/build/cli/index.js:173:3)
```

**Other Notes**

In searching for the keyword `WebSwapCGLLayer` within this repository, I found 6 different matches but most of them are not related to this circumstance (eg. one was for a chromium instance launched persistently, which this is not). 

The closest match is https://github.com/microsoft/playwright/issues/14687, which mentions https://github.com/electron/electron/issues/33685. 

I've also found other mentions of `WebSwapCGLLayer` in Chromium and Angleproject bug trackers:
- https://bugs.chromium.org/p/angleproject/issues/detail?id=7123
- https://bugs.chromium.org/p/angleproject/issues/detail?id=5253
- https://bugs.chromium.org/p/chromium/issues/detail?id=1217851

There appear to be no solution specifically mentioned in any of these existing bugs. [A StackOverflow post](https://stackoverflow.com/questions/72281193/class-webswapcgllayer-is-implemented-in-both-mac-os-x-system-and-node-modules) dismisses this as merely a warning, but in my case this crash prevents rest of the test suite from running.

Interestingly, the code snippet **will pass** if run in headless mode.

**Comment by worldomonation** [comment]
Sorry for the title gore.

**Comment by dgozman** [comment]
I think `WebSwapCGLLayer` is unrelated, the real issue is with the observers check. Merging into #16748 that will be fixed in v1.26.

**Comment by worldomonation** [comment]
Thanks, I will wait for 1.26 release :)
